outlines/generate/regex.py

```diff
@@ -35,8 +37,33 @@ def regex(model, regex_str: str, sampler: Sampler = multinomial()):
     return generator
 
 
+@regex.register(LlamaCpp)
+def regex_llamacpp(
+    model: LlamaCpp,
+    regex_str: str,
+    sampler: Sampler = multinomial(),
+):
+    if not isinstance(sampler, multinomial):
+        raise NotImplementedError(
+            r"The llama.cpp integration does not currently support any other sampling algorithm "
+            + "that the multinomial sampler."
+        )
+
+    logits_processor = RegexLogitsProcessor(regex_str, model.tokenizer)
+    model.logits_processor = logits_processor
+
+    fsm = RegexFSM(regex_str, model.tokenizer)
+    model.fsm = fsm
```

Comment:
Why having to pass a `RegexFSM` and a `LogitsProcessor`?

Reply:
Was this comment on an older version?